pal/image/minigame/image-data.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ Copyright (c) 2023 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+import { ALIPAY, XIAOMI, JSB, BAIDU, TAOBAO, TAOBAO_MINIGAME, WECHAT_MINI_PROGRAM } from 'internal:constants';
+import { BaseImageData } from '../base-image-data';
+import { ImageSource } from '../types';
+import { ccwindow } from '../../../cocos/core/global-exports';
+import { getError } from '../../../cocos/core';
+
+export class ImageData extends BaseImageData {
+    protected isNativeImage (imageSource: ImageSource): imageSource is (HTMLImageElement | HTMLCanvasElement | ImageBitmap) {
+        if (ALIPAY || TAOBAO || TAOBAO_MINIGAME || XIAOMI || BAIDU || WECHAT_MINI_PROGRAM) {
+            // We're unable to grab the constructors of Alipay native image or canvas object.
+            return !('_data' in imageSource);
+        }
+        return super.isNativeImage(imageSource);
+    }
+
+    set src (url: string) {
+        (this._imageSource as HTMLImageElement).src = url;
+    }
+
+    static downloadImage (url: string,
+        options: Record<string, any>,
+        onComplete: ((err: Error | null, data?: HTMLImageElement | null) => void)): HTMLImageElement {
```

Comment:
If return `HTMLImageElement `, then the invoker will use `HTMLImageElement`. And we want to avoid using it in engine codes except pal.

Reply:
This is designed for compatibility.

Reply:
Does it exported to developers? If not, then just modify engine codes.

Reply:
Yep, this is an export interface.
![image](https://github.com/cocos/cocos-engine/assets/5205637/c639845e-32ac-4341-991b-0ef726073634)

Reply:
I can not search it in https://docs.cocos.com/creator/api/zh/result?keyword=downloadImage.